machine-learning-scikit-learn.ipynb

```diff
@@ -0,0 +1,333 @@
+    "accuracy = 1 - errors.mean()\n",
+    "accuracy.compute()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inc.score(X_test, y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Grid search"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This accuracy might be the best possible, but there a lot of parameters to feed `SGDClassifier` that can drastically change the performance. We'll consider two of the most basic parameters, specifying a different loss function and different amount of penalty. This is the problem of hyperparameter optimization, something Dask-ML supports: https://dask-ml.readthedocs.io/en/latest/hyper-parameter-search.html\n",
+    "\n",
+    "One algorithm that Dask-ML implements a grid search algorithm through `GridSearchCV`. This algorithm takes in a list of values to evaluate and score each model at every possible combination on values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask_ml.model_selection import GridSearchCV\n",
+    "params = {'alpha': np.logspace(-6, 0, num=10),\n",
+    "          'loss': ['hinge', 'log', 'modified_huber', 'squared_hinge', 'perceptron']}\n",
+    "grid = GridSearchCV(est, params, return_train_score=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice that we pass the scikit-learn estimator here. `Incremental` is only a small wrapper for data management, and `GridSearchCV` is also a dask tool that for data management."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "# took about 3 minutes with 8 workers\n",
+    "grid.fit(X, y)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grid.best_params_"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the best loss, let's see how the score changed over different regularization constants `alpha`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.DataFrame(grid.cv_results_)\n",
+    "show = df[df.param_loss == grid.best_params_['loss']]\n",
+    "show.plot(x='param_alpha', y='mean_test_score', yerr='std_test_score',\n",
+    "          logx=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "But what if other losses were equally good? Let's see a visualization of the different losses *and* different regularization parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import seaborn as sns\n",
```

Comment:
I slightly prefer 

```python
g = sns.factorplot(x='param_alpha', y='mean_test_score', data=df,
                   aspect=3,
                   hue='param_loss')
```

for visualizing these.

![gh](https://user-images.githubusercontent.com/1312546/42125075-4307149c-7c35-11e8-9b8b-29e576f73d78.png)


ticklabels would need to be cleaned up.

will also need to ensure seaborn is available on the binder worker.

Reply:
I greatly prefer this instead of the heatmap.